keps/sig-cloud-provider/991-cloud-controller-migration/README.md

```diff
@@ -163,29 +169,50 @@ type ControllerLeaderConfiguration struct {
 	// E.g. kube-controller-manager, cloud-controller-manager, etc
 	Component string `json:"component"`
 }
 ```
+#### Default LeaderMigrationConfiguration
+
+The `controller-manager` will provide `kube-controller-manager` and `cloud-controller-manager`
+each a default `LeaderMigrationConfiguration` that represents the situation where the controller manager is running with
+default assignments of controllers and lock type selection.
+
+The default values must be only used when no configuration file is specified. If a custom configuration file is
+specified to either controller manager, the specified configuration will completely replace default value for the
+corresponding controller manager.
 
 #### Component Flags
 
-The LeaderMigrationConfiguration type will be read by the `kube-controller-manager` and the `cloud-controller-manager` via a new flag `--cloud-migration-config` which
-accepts a path to a file containing the LeaderMigrationConfiguration type in yaml.
+Both `kube-controller-manager` and `cloud-controller-manager` will add support for the following two flags for Leader
+Migration. First, `--enable-leader-migration` is a boolean flag which defaults to `false` that indicates whether
+Leader Migration is enabled. Second, `--leader-migration-config` is an optional flag that accepts a path to a file
+containing the `LeaderMigrationConfiguration` type serialized in yaml.
+
+If `--enable-leader-migration` is `true` but `--leader-migration-config` flag is empty or not set, the
+default `LeaderMigrationConfiguration` for corresponding controller manager will be used.
+
+If `--enable-leader-migration` is not set or set to `false`, but `--leader-migration-config` is set and not empty, the
+controller manager will print an error at `FATAL` level and exit immediately.
+
+If `--leader-migration-config` is set but the configuration file cannot be read or parsed, the controller manager will
+log the failure at `FATAL` level and exit immediately.
 
 #### Example Walkthrough of Controller Migration
 
 This is an example of how you would migrate all cloud controllers from the CCM to the KCM during a typical cluster version upgrade.
 
 ##### Enable Leader Migration on Components
 
-First, define a LeaderMigrationConfiguration resource in a yaml file containing all known cloud controllers. The component name for each controller should be set to
-the component where the controllers are currently running. Almost always this is the `kube-controller-manager`. The configuration file should look something like this:
+First, define a LeaderMigrationConfiguration resource in a yaml file containing all known cloud controllers. The component name for each controller must be set to
+the component where the controllers are currently running. Almost always this is the `kube-controller-manager`. The configuration file is something like this:
 ```yaml
 kind: LeaderMigrationConfiguration
 apiVersion: v1alpha1
 leaderName: cloud-controllers-migration
```

Comment:
Can we fix this leaderName, I don't think this is a leaderName we should ever use. One of the following would seem much better.
CloudExtractionMigration
KCMToCCMMigration
1-21To1-22Migration

Reply:
I think the version number is unlikely important because we support only upgrade from one version to its direct next.